frontend/public/components/factory/list.tsx

```diff
@@ -565,6 +568,7 @@ export type ListInnerProps = {
   data?: any[];
   defaultSortField?: string;
   defaultSortFunc?: string;
+  NoDataMsg?: React.ComponentType<{}>;
```

Comment:
Instead of `NoDataMsg` and `EmptyMsg`, I'd recommend

* `EmpyMsg` -- no data exists
* `AllItemsFilteredMsg` -- all items are hidden by filters

Then we can fallback to `EmptyMsg` if `AllItemsFilteredMsg` is not provided.